crates/polars-ops/src/series/ops/round.rs

```diff
@@ -37,6 +38,28 @@ pub trait RoundSeries: SeriesSealed {
         polars_bail!(opq = round, s.dtype());
     }
 
+    fn round_sig_figs(&self, digits: i32) -> PolarsResult<Series> {
+        let s = self.as_series();
+        if digits < 1 {
+            polars_bail!(
+                InvalidOperation: "digits must be an integer >= 1"
+            )
+        };
+
+        with_match_physical_numeric_polars_type!(s.dtype(), |$T| {
+            let s = s.cast(&DataType::Float64)?;
+            let ca = s.f64()?;
+            let s = ca.apply_values(|value| {
+                if value == 0.0 {
+                    return 0.0;
+                }
+                let magnitude = 10.0_f64.powi(digits - 1 - ((value.abs()).log10().floor() as i32));
+                (value * magnitude).round() / magnitude
+            }).into_series();
+            return Ok(s);
+        });
```

Comment:
Hi @orlp sorry its been a bit messy! I have gotten this to work with the type matching here. Let me know 1. if this is ok - not sure if the explicit casting to float64 in the middle is acceptable and 2. I am not sure how to convert back to the original datatype using $T - can you advise? Thank you.

Reply:
Also this is slower than the original implementation I had - still about 10% faster than the original numpy solution so not a big problem.